### PR TITLE
Fix scylla bench

### PR DESCRIPTION
**Pull request by dkropachev**
Addresses following problems:
1. Stop installing scylla-bench when it is not needed
2. Fix scylla-bench installation issue
3. Prop up golang version

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [ ] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (CI)
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

File: sdcm/cluster.py [hunk 1]
```diff
@@ -4653,12 +4655,12 @@ def install_scylla_bench(node):
             """))
         node.remoter.sudo(shell_script_cmd("""\
             rm -rf /usr/local/go
-            curl -LO https://storage.googleapis.com/golang/go1.13.linux-amd64.tar.gz
+            curl -LO https://storage.googleapis.com/golang/go1.16.3.linux-amd64.tar.gz
             tar -C /usr/local -xvzf go1.13.linux-amd64.tar.gz
```
Comment (aleksbykov) [inline sdcm/cluster.py:4659]: SHould not it be changed here also?

File: sdcm/sct_config.py [hunk 2]
```diff
@@ -1456,18 +1472,19 @@ def _validate_value(self, opt):
             cur_val = self.get(opt['name'])
             assert cur_val in choices, "failed to validate '{}': {} not in {}".format(opt['name'], cur_val, choices)
 
+    @property
+    def list_of_stress_tools(self) -> List[str]:
+        output = []
```
Comment (bentsi) [inline sdcm/sct_config.py:1477]: let's decrease nesting by:

```
stress_tools = set()
...
stress_tools.add(stress_tool)
```

File: sdcm/sct_config.py [hunk 3]
```diff
@@ -1215,6 +1215,22 @@ class SCTConfiguration(dict):
...
     ]
 
+    stress_cmd_params = [
+        # this list is used for variouse checks against stress commands, such as:
+        # 1. Check if all c-s profile files existing that are referred in the commands
+        # 2. Check what stress tools test is needed when loader is prepared
+        'stress_cmd', 'stress_read_cmd', 'stress_cmd_w', 'stress_cmd_r', 'stress_cmd_m',
+        'prepare_write_cmd', 'stress_cmd_no_mv', 'stress_cmd_no_mv_profile',
+        'prepare_stress_cmd', 'stress_cmd_1', 'stress_cmd_complex_prepare', 'prepare_write_stress',
+        'stress_cmd_read_10m', 'stress_cmd_read_cl_one', 'stress_cmd_read_80m',
```
Comment (aleksbykov) [inline sdcm/sct_config.py:1225]: I think gemini also should be here: `gemini_cmd`

File: test-cases/performance/perf-row-level-repair-1TB.yaml [hunk 4]
```diff
@@ -9,6 +9,8 @@ prepare_write_cmd: ["cassandra-stress write no-warmup cl=ALL n=125000000 -schema
                     "cassandra-stress write no-warmup cl=ALL n=125000000 -schema 'replication(factor=3)' -port jmx=6868 -mode cql3 native -rate threads=200 -col 'size=FIXED(1024) n=FIXED(1)' -pop seq=750000001..875000000",
                     "cassandra-stress write no-warmup cl=ALL n=125000000 -schema 'replication(factor=3)' -port jmx=6868 -mode cql3 native -rate threads=200 -col 'size=FIXED(1024) n=FIXED(1)' -pop seq=875000001..1000000000"]
 
+stress_cmd: "scylla-bench -workload=sequential -mode=write -max-rate=300 -replication-factor=3 -concurrency=10 -rows-per-request=30"
```
Comment (bentsi) [inline test-cases/performance/perf-row-level-repair-1TB.yaml:12]: why did you add it to this test case?

File: performance_regression_row_level_repair_test.py [hunk 5]
```diff
@@ -297,9 +297,7 @@ def test_row_level_repair_large_partitions(self):
         clustering_row_size = 2048
         partition_count_per_node = partition_count // 10 // 3
         clustering_row_count_per_node = clustering_row_count // 100 * 3
-        scylla_bench_base_cmd = "scylla-bench -workload=sequential -mode=write -max-rate=300 " \
-                                "-replication-factor=3 -clustering-row-size={} -concurrency=10 -rows-per-request=30".format(
-                                    clustering_row_size)
+        scylla_bench_base_cmd = self.params.get('stress_cmd') + f" -clustering-row-size={clustering_row_size}"
```
Comment (bentsi) [inline performance_regression_row_level_repair_test.py:300]: This test is  relying on Scylla Bench, assert if the stress command is different

File: test-cases/performance/perf-row-level-repair-1TB.yaml (hunk 4 above)
Comment (dkropachev) [reply test-cases/performance/perf-row-level-repair-1TB.yaml:12]: This test relying on scylla-bench, so it needs to tell sct to install it

File: sdcm/sct_config.py (hunk 3 above)
Comment (dkropachev) [reply sdcm/sct_config.py]: fixed

File: sdcm/sct_config.py (hunk 2 above)
Comment (dkropachev) [reply sdcm/sct_config.py]: fixed

File: sdcm/cluster.py (hunk 1 above)
Comment (dkropachev) [reply sdcm/cluster.py]: thanks, fixed

File: performance_regression_row_level_repair_test.py (hunk 5 above)
Comment (dkropachev) [reply performance_regression_row_level_repair_test.py:302]: Added assertion

**Review by bentsi — APPROVED** [review]

**Comment by roydahan** [comment]
@bentsi please backport to all affected versions.
I need it in 4.5 but if it's going to fail runs also in 4.3 and 4.4 and possibly 2020/1 we will need it there.
